wg-structured-logging/charter.md

```diff
@@ -0,0 +1,70 @@
+# WG Structured Logging Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README]
+and uses the Roles and Organization Management outlined in [sig-governance].
+
+[sig-governance]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md
+[Kubernetes Charter README]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/README.md
+
+## Scope
+
+Modernize logging in Kubernetes core components, allowing users to efficiently consume, process, store and analyse 
+information stored in logs.
+
+### In Scope
+
+- Define the standard for logging - propose libraries, interfaces, metadata schema
+- Reduce friction for using logging - reduce dependencies and performance overhead
+- Give more choice over logging - allow pluggable logging implementation
+- Ensure quality consistent logging - overview migration, create documentation, tooling and educate reviewers
+- Prevent regressions caused by logging - measure performance overhead and log volume changes
+
+For all the above points, we will focus on core Kubernetes components and addons.
+Other SIG subprojects/components (e.g. SIG Scheduling descheduler) are out of
+scope. However, we expect that tooling created by the group should be easy to adapt more broadly.
+
+### Out of scope
+
+- Logging of non-core Kubernetes components
```

Comment:
where does logging done by upstream dependencies fit? is everything that contributes to the logfile of core Kubernetes components in-scope if the entire log file is expected to be structured?

Reply:
Yes, I think that covering upstream dependencies is needed to have logs in single format which is required to achieve goal of making logs easy to consume. 

To make sure I understood you correctly, when talking about "upstream" dependencies do you mean:
* Kubernetes shared libraries in ./staging/src/k8s.io directory? or
* Third party libraries outside of k/k repo?

Reply:
ping @liggitt

Reply:
> To make sure I understood you correctly, when talking about "upstream" dependencies do you mean:
> 
> * Kubernetes shared libraries in ./staging/src/k8s.io directory? or
> * Third party libraries outside of k/k repo?

I mean both:
1. things inside the staging/src/k8s.io
2. things in k8s.io/kubernetes/vendor that log or output to os.Stdout / os.Stderr

It seems like all have to be addressed to make the core components output completely structured... number 2 is a significantly bigger task/scope

Reply:
I don't think that number 2 is as big as it might seem at first sight. Issues with third party libraries were already present before structured logging effort. klog also standardizes a log output format by using a standard header. For example we are injecting a klog based implementation of zap logger so Etcd client will write logs in klog format.

We already don't provide any guarantee that all logs have klog format, so the question should be "are we ok with the current state of log format completeness or should we do any steps to improve it?"

As charter specifies wg should allow "users to efficiently consume (logs)". WG should measure impact of third party logging on quality of log output and improve it if needed, but I don't it will be easy or profitable to target 100% coverage.

Reply:
ok, maybe clarify this out-of-scope bit to "Logging outside of the kubernetes/kubernetes repository", since this WG does care about logging by things inside kubernetes/kubernetes vendor

Reply:
Updated, this will additionally add the other non-core components like kubeadm, but I think we should have uniform logging withing k/k.

Reply:
oh, that's a good point... I was focused on the impacts of logging from dependencies of the core components

the considerations for CLI components like kubeadm and kubectl are different... I see you described those as being out of scope in https://github.com/kubernetes/community/pull/5770/files#r644257373 ... would copying that line into the "out of scope" section and including kubeadm help clarify the WG focus?

> Binaries like kubectl and kubeadm will not be directly targeted nor we plan to change application logs that kubectl logs reads.

Reply:
we also saw a number of structured logging PRs for /test.